dev/TeachingTip/TeachingTip.xaml

```diff
@@ -184,6 +184,29 @@
                                         <Setter Target="TailPolygon.Margin" Value="{StaticResource TeachingTipTailPolygonMarginLeft}"/>
                                     </VisualState.Setters>
                                 </VisualState>
+                                <VisualState x:Name="LeftRTL">
+                                    <VisualState.Setters>
```

Comment:
Having these "RTL" states seems overly complicated, we shouldn't need these if the mirroring is doing its job correctly. I think what you're saying is that when the popup's child has a flow direction different than the window root, you have to compensate. So I would have expected to see another node inserted between the popup child and the teachingtip's content that serves that purpose. I think if we did that then we wouldn't need these extra VSM states.

Additionally, I think it is a non-scenario to handle the case where the TeachingTip and the target have different flow direction. If that happens and the tail is on the "wrong side", i think the developer will understand. I don't want to add this much extra complexity to the control to handle a super niche case.

Reply:
The flow direction of the target doesn't actually matter, if the content of the tip is supposed to be RTL we run into issues because the popup root is always LTR.  
I agree there is probably a place we could insert a new grid into the tree in order to get the the content to have the correct FlowDirection while maintaining the tip chrome which is being layed out in LTR (unconditionally, due to the Parentless Popup behavior I described in the description).
The tip's template is very carefully crafted to get the tail edge border to look appropriate, so the highlight and border color track to the edge of the tail. So it may be tricky to get this just right.